### ISSUE TITLE
Basic Auth in command line

### ISSUE THREAD

**Issue by WTFKr0** [containous/traefik#885]
Hi !

Today i run traefik only on command line, without config file.
I would add the AuthBasic feature but i could'nt find option in "traefik --help" on traefik:camembert
Is the option absent and only available in config file ?

Section in config file that I want in command line : 
```
#   [web.auth.basic] 
#     users = ["test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/","test2:$apr1$d9hr9HBB$4HxwgUir3HP4EsggP/QNo0"]
```

Thanks

**Comment by WTFKr0** [comment]
And that too : 
```
# [entryPoints]
#   [entryPoints.http]
#   address = ":80"
#   [entryPoints.http.auth.basic]
#   users = ["test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/","test2:$apr1$d9hr9HBB$4HxwgUir3HP4EsggP/QNo0"]
``` 

I think it will be complicated to add that in the : 
```
--entrypoints                                      Entrypoints definition using format: --entryPoints='Name:http Address::8000 Redirect.EntryPoint:https' --entryPoints='Name:https Address::4442 TLS:tests/traefik.crt,tests/traefik.key' (default "map[]")
```

**Comment by emilevauge** [comment]
@WTFKr0 I think that was the reason why it was not added initially... We have a problem transposing to arguments complex configuration objects like entrypoints...

**Comment by WTFKr0** [comment]
Yeah
Maybe it's more a flaeg issue.
How to multiple values (map) with multiple subparameters.
Here the entrypoint id is the Name parameter, so we can imagine that flaeg allow something like : 

--entryPoints.http.Address="8000"
--entryPoints.https.Address="4442"
--entryPoints.https.Address="4442"
--entryPoints.https.TLS="tests/traefik.crt,tests/traefik.key"

Like this, we can easily add a param to map values
This method reminds me the "service" model of frontend labels in #444

**Comment by emilevauge** [comment]
@WTFKr0 indeed, we should use the same strategy as in #444 :)
In the meantime, please use a toml file to add basic auth.

**Comment by kleinsasserm** [comment]
Yes, that would be nice as said in #444. I would like to request this because it is easier to handle the arguments with systemctl environmentfile.

**Comment by jmagnusson** [comment]
Would be nice to have a more dynamic way than through the main config file, to supply the top-level users/passwords for the web UI.

**Comment by kachkaev** [comment]
Now that access to web UI and API is configured via an entrypoint, can the issue be considered as resolved?

I guess you can do something like:

```bash
traefik \
  --entryPoints='Name:http Address::80 Redirect.EntryPoint:https' \
  --entryPoints='Name:https Address::443 TLS' \
  --entryPoints='Name:traefik Address::8080 Auth.Basic.Users:test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/,test2:$apr1$d9hr9HBB$4HxwgUir3HP4EsggP/QNo0' \
  # ...other arguments
```

All available entrypoint options: https://docs.traefik.io/configuration/entrypoints/#all-available-options

API entrypoint definition: https://docs.traefik.io/configuration/api/

**Comment by ldez** [comment]
Closed by #2799
